modules/gapi/perf/cpu/gapi_video_perf_tests_cpu.cpp

```diff
@@ -0,0 +1,57 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+
+#include "../perf_precomp.hpp"
+
+#include "../common/gapi_video_perf_tests.hpp"
+#include <opencv2/gapi/cpu/video.hpp>
+
+namespace
+{
+#define VIDEO_CPU cv::gapi::video::cpu::kernels()
+} // namespace
+
+namespace opencv_test
+{
+
+INSTANTIATE_TEST_CASE_P(OptFlowLKPerfTestCPU, OptFlowLKPerfTest,
+                        Combine(Values(AbsExactForOptFlow().to_compare_f()),
+                                Values("cv/optflow/rock_%01d.bmp",
+                                       "cv/optflow/frames/1080p_%02d.png"),
+                                Values(1, 3, 4),
+                                Values(std::make_tuple(9, 9), std::make_tuple(15, 15)),
+                                Values(7, 11),
+                                Values(cv::TermCriteria(cv::TermCriteria::COUNT |
+                                                        cv::TermCriteria::EPS,
+                                                        30, 0.01)),
+                                Values(cv::compile_args(VIDEO_CPU))));
+
+INSTANTIATE_TEST_CASE_P(OptFlowLKForPyrPerfTestCPU, OptFlowLKForPyrPerfTest,
+                        Combine(Values(AbsExactForOptFlow().to_compare_f()),
+                                Values("cv/optflow/rock_%01d.bmp",
+                                       "cv/optflow/frames/1080p_%02d.png"),
+                                Values(1, 3, 4),
+                                Values(std::make_tuple(9, 9), std::make_tuple(15, 15)),
+                                Values(7, 11),
+                                Values(cv::TermCriteria(cv::TermCriteria::COUNT |
+                                                        cv::TermCriteria::EPS,
+                                                        30, 0.01)),
+                                Values(true, false),
+                                Values(cv::compile_args(VIDEO_CPU))));
+
+INSTANTIATE_TEST_CASE_P(OptFlowLKInternalPerfTestCPU, OptFlowLKForPyrPerfTest,
+                        Combine(Values(AbsExactForOptFlow().to_compare_f()),
+                                Values("cv/optflow/rock_%01d.bmp"),
+                                Values(1),
+                                Values(std::make_tuple(10, 10)),
+                                Values(15),
+                                Values(cv::TermCriteria(cv::TermCriteria::COUNT |
+                                                        cv::TermCriteria::EPS,
+                                                        21, 0.05)),
+                                Values(true),
+                                Values(cv::compile_args(VIDEO_CPU))));
```

Comment:
And this code actually depends on the presense of OPENCV_VIDEO module.

If this module is not built, these tests shouldn't be instantiated (or disabled with a macro as @Volskig did that)

Here I see tests are instantiated always so in some (pretty weird) configurations tests will fail.

Reply:
disabled by #ifdef macro (DISABLED_ would turn off it always)

Reply:
The idea was to put `DISABLED_` optionally.

```c++
#ifdef HAVE_OPENCV_VIDEO
#define WITH_VIDEO(X) X
#else
#define WITH_VIDEO(X) DISABLED_##X 
#endif // HAVE_OPENCV_VIDEO

INSTANTIATE_TEST_CASE_P(WITH_VIDEO(OptFlowLKInternalPerfTestCPU), ...
```

Reply:
Oh, now I get it. Done
Also anticipate questions: `INSTANTIATE_TEST_CASE_P()` macro concatenates the arguments in its body, therefore prescan to seek macros in arguments does not occur; that's why I have to use another macro which calls gtest-macro with macro in the first argument to expand it properly

Reply:
Ok, it also works, thanks!